Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -155,14 +155,32 @@ template <class Type> class TableColumn : public API::Column {
    * is throw. In case of an overflow boost::numeric::positive_overflow or
    * boost::numeric::negative_overflow
    * is throw.
-   * @param i :: The index to an element.
+   * @param value :: The value of the element.
    */
-  double toDouble(size_t i) const override {
+  template <typename T> double convertToDouble(const T &value) const {
     typedef
```

Comment:
To put this into context, we allow `V3D` in TableColumns, see [here](https://github.com/mantidproject/mantid/blob/master/Framework/DataObjects/src/TableColumn.cpp#L16), hence we have to defer the error to runtime (should this method with a `V3D` ever be called).